packages/i18nify-js/src/modules/phoneNumber/__tests__/getMaskedPhoneNumber.test.ts

```diff
@@ -0,0 +1,139 @@
+import { CountryCodeType } from '../../types';
+import { getMaskedPhoneNumber, GetMaskedPhoneNumberOptions } from '../index';
+
+describe('phoneNumber - getMaskedPhoneNumber', () => {
+  it('should throw error if no countryCode and phoneNumber are provided', () => {
+    expect(() =>
+      getMaskedPhoneNumber({} as GetMaskedPhoneNumberOptions),
+    ).toThrow('Either countryCode and phoneNumber is mandatory.');
+  });
+
+  it('should throw error for invalid country code when phone number is not provided', () => {
+    expect(() =>
+      getMaskedPhoneNumber({ countryCode: 'ZZ' as CountryCodeType }),
+    ).toThrow('Parameter "countryCode" is invalid: ZZ');
+  });
+
+  it('should throw error for invalid country code when phone number is provided', () => {
+    const options = {
+      countryCode: 'ZZ' as CountryCodeType,
+      phoneNumber: '7394926646',
+    };
+    const expected = 'xxxxxxxxxx';
+    expect(getMaskedPhoneNumber(options)).toEqual(expected);
+  });
+
+  it('should return full masked phone number with dial code', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '+12345678901',
+      withDialCode: true,
+      maskingOptions: { maskingStyle: 'full' },
+    };
+    const expected = '+1 xxx-xxx-xxxx';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should return phone number masked with prefix', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: false,
+      maskingOptions: { maskingStyle: 'prefix', maskedDigitsCount: 6 },
+    };
+    const expected = 'xxx-xxx-8901';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should return phone number masked with suffix', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: false,
+      maskingOptions: { maskingStyle: 'suffix', maskedDigitsCount: 6 },
+    };
+    const expected = '234-5xx-xxxx';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should handle alternate masking of digits', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: false,
```

Comment:
is maskingChar not default set to * ?

Reply:
Default value of `maskingChar` is `x`

<img width="233" alt="Screenshot 2024-05-14 at 10 10 25 AM" src="https://github.com/razorpay/i18nify/assets/61814592/a066fb9a-0c91-4707-af24-22ae29caacc0">